hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -188,6 +239,25 @@ protected boolean waitInitialized(MasterProcedureEnv env) {
     return am.waitMetaLoaded(this) || am.waitMetaAssigned(this, getRegion());
   }
 
+  private void checkAndWaitForOriginalServer(ServerName lastHost)
+    throws ProcedureSuspendedException {
+    ServerName newNameForServer = serverManager.findServerWithSameHostnamePortWithLock(lastHost);
+    boolean isOnline = serverManager.createDestinationServersList().contains(newNameForServer);
+    if (!isOnline && retries < forceRegionRetainmentRetries) {
+      retries++;
+      LOG.info("Suspending the TRSP PID={} because {} is true and previous host {} "
+        + "for region is not yet online.", this.getProcId(), FORCE_REGION_RETAINMENT, lastHost);
+      setTimeout(forceRegionRetainmentWait);
```

Comment:
So the `forceRegionRetainmentWait` here is interval? Not total time?

Reply:
Yes, the wait interval between attempts, not the total time.